cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -124,7 +124,8 @@ public void writeProfile(
     Profile profile =
       new Profile(profileName, profileCreateRequest.getDescription(),
                   profileId.getNamespaceId().equals(NamespaceId.SYSTEM) ? EntityScope.SYSTEM : EntityScope.USER,
-                  profileCreateRequest.getProvisioner());
+                  new ProvisionerInfo(profileCreateRequest.getProvisioner().getName(),
```

Comment:
I think it's better to do this in ProfileCreateRequest's getProvisioner() method, to ensure it always returns a ProvisionerInfo without null properties.

Reply:
fixed